src/detect-engine-analyzer.c

```diff
@@ -417,6 +418,8 @@ void CleanupRuleAnalyzer(void)
         fclose(rule_engine_analysis_FD);
         rule_engine_analysis_FD = NULL;
     }
+    pcre2_code_free(percent_re);
```

Comment:
Even though both functions can handle null pointers, i'd suggest doing a null check before invoking each.

Reply:
Are there other places with the same thing ?